cosmwasm/contracts/global-accountant/tests/helpers/mod.rs

```diff
@@ -67,39 +69,33 @@ impl Contract {
         )
     }
 
-    pub fn modify_balance(
-        &mut self,
-        modification: Binary,
-        guardian_set_index: u32,
-        signatures: Vec<Signature>,
-    ) -> anyhow::Result<AppResponse> {
-        self.app.execute_contract(
-            Addr::unchecked(USER),
+    pub fn modify_balance(&mut self, modification: Binary) -> anyhow::Result<AppResponse> {
+        let migrate_msg = MigrateMsg {
+            modifications: vec![ModifyBalance { modification }],
+        };
+        // let serialized_msg = serde_json::to_string(&migrate_msg).unwrap();
+        // let serialized_msg = migrate_msg.serialize()?;
+        // since migrate is an authenticated method, we reuse
+        // that with the same code_id to pass sensitive messages
+        // like ModifyBalance
+        self.app.migrate_contract(
+            Addr::unchecked(ADMIN),
             self.addr(),
-            &ExecuteMsg::ModifyBalance {
-                modification,
-                guardian_set_index,
-                signatures,
-            },
-            &[],
+            &migrate_msg,
+            self.code_id,
         )
```

Comment:
It feels a bit weird to me that the only way to modify an account's balance is to migrate the entire contract.  The two things are kind of orthogonal.

Reply:
I'm open for further discussion on this.  Adding new messages for guardians to sign is sensitive & can add operationally overhead.  The migrate + instantiate are kind of secure channels we can reuse conveniently as a pattern, but I agree it seems weird.  At least in the migration governance message it will be clear that the code_id is staying the same, and it's the balances getting "migrated".